source/simulator/assembly.cc

```diff
@@ -748,18 +750,18 @@ namespace aspect
   }
 
   template <int dim>
+  template <typename T>
   void
   Simulator<dim>::
-  get_artificial_viscosity (Vector<float> &viscosity_per_cell,
-                            const AdvectionField &advection_field) const
+  get_artificial_viscosity (Vector<T> &viscosity_per_cell,
+                            const AdvectionField &advection_field,
+                            const bool use_smoothing /* = false */) const
   {
     Assert(viscosity_per_cell.size()==triangulation.n_active_cells(), ExcInternalError());
 
     if (advection_field.field_type == AdvectionField::compositional_field)
       Assert(parameters.n_compositional_fields > advection_field.compositional_variable, ExcInternalError());
 
-    viscosity_per_cell = 0.0;
-
```

Comment:
Or is this because you know that you're going to write into every cell's element of the vector at least once anyway?